napari/layers/points/_points_key_bindings.py

```diff
@@ -58,7 +60,61 @@ def paste(layer: Points):
     trans._("Select all points in the current view slice."),
 )
 def select_all(layer: Points):
-    layer.selected_data = set(layer._indices_view[: len(layer._view_data)])
+    new_selected = set(layer._indices_view[: len(layer._view_data)])
+
+    # If all visible points are already selected, deselect the visible points
+    if new_selected & layer.selected_data == new_selected:
+        layer.selected_data = layer.selected_data - new_selected
+        show_info(
+            trans._(
+                "Deselected all points in this slice,"
+                " use Shift-A to deselect all points on the layer."
+                f" ({len(layer.selected_data)} selected)"
```

Comment:
Unfortunately, due to a limitation of how translations work, the string *must* be in a whole line. Therefore, in this case, it's ok to make the line too long by putting the whole string together.

Reply:
Or does it only not work when the string is actually translated?

Reply:
```python
In [1]: from napari.utils.translations import trans

In [14]: deferred_string = trans._(
    ...:     "Hello"
    ...:     " {word}"
    ...:     ", This is"
    ...:     " {word2}",
    ...:     word="World",
    ...:     word2="Test",
    ...:     deferred=True,
    ...:     )

In [15]: deferred_string.value()
Out[15]: 'Hello World, This is Test'

In [16]: deferred_string.translation()
Out[16]: 'Hello World, This is Test'
```

Reply:
But I removed the f-strings already.

Reply:
The problem here is not how python treats strings but the script that collects strings for translation. This script reads text files and collects each line as a separate string that will remove the context needed in translation.

Reply:
Ok, do the format arguments also need to be in the same line?

Reply:
And in the overall code, sometimes deferred is provided as an argument and sometimes it is not and I missed to see the pattern when to use it.

Reply:
I concatenated the strings.

Reply:
> And in the overall code, sometimes deferred is provided as an argument and sometimes it is not and I missed to see the pattern when to use it.

https://napari.org/developers/translations.html#deferred-translations

Reply:
@Czaki Thanks for the link and that I understand, however it is more about when is the deferred option appropriate and when it is not? Or should it always be?

So I tried to learn from the code but I cannot really say: In case of a variable string assignment I use deferred and otherwise I don't use it like:
It is used here as input to the ValueError:

https://github.com/napari/napari/blob/458a7e0f0a15c17fab12cb5ae45cbfc591fa303e/napari/utils/validators.py#L63-L70

but not here:

https://github.com/napari/napari/blob/458a7e0f0a15c17fab12cb5ae45cbfc591fa303e/napari/utils/migrations.py#L25-L30

Since I want to use it correctly, my question would be when to use it and if my current use case falls into this category. 
From my understanding I would say, that I should use deferred when I want to evaluate the translation later manually, but wouldn't that mean that if I give a deferred=True as input to the ValueError that it would never be translated, because `str(deferred_string)` is the untranslated version and needs an explicit call to `.translate()`?

Sorry if I sound confused, but I see `deferred=True` all over the code as inputs to Exceptions and wonder if I did not understand the content of the link you provided.

Reply:
It looks like in the second one I forgot (it is from my PR) about derefered as it may happen when using CLI.

Reply:
ok, so I will just always use it. :)